src/Util.cpp

```diff
@@ -62,19 +63,76 @@ string running_program_name() {
     #endif
 }
 
-// TODO: Rationalize the two different versions of unique_name,
-// possibly changing the name of one of them as they are used
-// for different things, and in fact the two versions can end
-// up returning the same name, thus they are not collectively
-// unique.
+namespace {
+const int num_unique_name_counters = (1 << 14);
+std::atomic<int> unique_name_counters[num_unique_name_counters];
+
+int unique_count(size_t h) {
+    h = h & (num_unique_name_counters - 1);
+    return unique_name_counters[h]++;
+}
+}
+
+// There are three possible families of names returned by the methods below:
+// 1) (char that isn't '$') + number (e.g. v234)
```

Comment:
I would go ahead and name the patterns here. E.g.:
    `// 1) char pattern: (char that isn't '$') + number (e.g. v234)`

The names match those used below.